sklearn/semi_supervised/tests/test_label_propagation.py

```diff
@@ -26,13 +27,27 @@
     }),
 ]
 
+ESTIMATOR_CLS = [label_propagation.LabelPropagation,
+                 label_propagation.LabelSpreading]
 
-def test_fit_transduction():
+
+PARAMETERS = [{'kernel': 'rbf'}, {'kernel': 'knn', 'n_neighbors': 2}, {
+        'kernel': lambda x, y: rbf_kernel(x, y, gamma=20)
+    }]
+
+
+MATRIX_TYPES = (sp.csc_matrix, sp.csr_matrix, sp.coo_matrix, sp.dok_matrix,
+                sp.bsr_matrix, sp.lil_matrix, sp.dia_matrix, np.array)
```

Comment:
I agree with your point and this setup: using test parametrization definitely would be better. :+1: 

You can proceed in two steps in different commits (at least two but not necessarily) to:
 - adapt the other tests to assert the correct behavior of `fit` on sparse matrices
 - if all the tests pass, adapt the test to use parametrisation as you started to do with `test_fit_transduction`
 
If we judge that it would be better having a dedicated PR for this second step, we can discard them here but cherry-pick them on another PR branch. :slightly_smiling_face:

Reply:
I did this due the reviewer comments in the stalled PR that this PR takes-over
https://github.com/scikit-learn/scikit-learn/pull/17384#discussion_r454514567

I agree that we should have a dedicated PR instead. I will revert these edits and only focus only on the main the tests/funcs.